vmm/src/device_manager.rs

```diff
@@ -509,6 +509,145 @@ pub enum DeviceManagerError {
     InvalidConsoleFd,
 }
 
+impl fmt::Display for DeviceManagerError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
```

Comment:
I don't like this filled in with `todo!()` statements. I'll comment on the original PR with an approach to move forward.

Reply:
That was exactly my point above!